src/devices/Vl53L0X/Vl53L0X.cs

```diff
@@ -0,0 +1,1163 @@
+                    throw new IOException($"{nameof(ReadContinuousMeasrurementMillimeters)} timeout error");
+            }
+            // assumptions: Linearity Corrective Gain is 1000 (default)
+            // fractional ranging is not enabled
+            var range = ReadUInt16((byte)Registers.RESULT_RANGE_STATUS + 10);
+            WriteRegister((byte)Registers.SYSTEM_INTERRUPT_CLEAR, 0x01);
+            return range;
+        }
+
+        /// <summary>
+        /// Get the distance depending on the measurement mode
+        /// </summary>
+        public UInt16 Distance => MeasurementMode == MeasurementMode.Continuous ? DistanceContinousMillimeters : GetDistanceSingleMillimeters(true);
+
+        /// <summary>
+        /// Get/Set the measurement mode used to return the distance property
+        /// </summary>
+        public MeasurementMode MeasurementMode { get; set; }
+
+        /// <summary>
+        /// Get a distance in millimeters from the continous measurement feature.
+        /// It is recommended to used this method to gethigher quality measurements
+        /// </summary>
+        public UInt16 DistanceContinousMillimeters
+        {
+            get
+            {
+                if (!_continuousInitialized)
+                {
+                    StartContinuousMeasurement();
+                }
+                return ReadContinuousMeasrurementMillimeters();
+            }
+        }
+
+        /// <summary>
+        /// Get a distance in millimeters
+        /// </summary>
+        /// <param name="multipleReads">True if you want multiple try to get a clean value</param>
+        /// <returns>Returns the maximum range so 8190</returns>
+        public UInt16 GetDistanceSingleMillimeters(bool multipleReads = false)
+        {
+            try
+            {
+                var value = DistanceSingleMeasurementMillimeters;
+                // Sensor can read maximum range while there is an object in front
+                // Make an average with a few reading withg the good readings
+                // Catch any exception and return the maximum range
+                Double average = 0.0;
+                int goodCount = 0;
+                if (multipleReads)
+                {
+                    for (int i = 0; i < MaxTryReadSingle; i++)
+                    {
+                        if (value < (UInt16)OperationRange.OutOfRange)
+                        {
+                            goodCount++;
+                            average += value;
+                        }
+                        value = DistanceSingleMeasurementMillimeters;
+                    }
+
+                }
+                else
+                {
+                    return value;
+                }
+                return goodCount != 0 ? (UInt16)(average / goodCount) : (UInt16)OperationRange.OutOfRange;
+            }
+            catch (IOException)
+            {
+                return (UInt16)OperationRange.OutOfRange;
```

Comment:
mixed feelings about this but fine for now until we got units more stabilized

Reply:
The sensor has a maximum range and returns it when object is too far or when there is an issue